Jenkinsfile.rh.release

```diff
@@ -24,9 +24,44 @@ properties([
   ]),
 ])
 
+String version = ''
+String checksum = ''
+
+void updateIQServerVersionAndChecksum(String version, String checksum) {
+  def dockerFile = readFile(file: 'Dockerfile.rh')
+  def metaShortVersionRegex = /(release=")(\d\.\d{1,3}\.\d)(" \\)/
+  def versionRegex = /(ARG IQ_SERVER_VERSION=)(\d\.\d{1,3}\.\d\-\d{2})/
+  def shaRegex = /(ARG IQ_SERVER_SHA256=)([A-Fa-f0-9]{64})/
+  dockerFile = dockerFile.replaceAll(metaShortVersionRegex,
+      "\$1${version.substring(0, version.indexOf('-'))}\$3")
+  dockerFile = dockerFile.replaceAll(versionRegex, "\$1${version}")
+  dockerFile = dockerFile.replaceAll(shaRegex, "\$1${checksum}")
+
+  writeFile(file: 'Dockerfile.rh', text: dockerFile.rh)
+}
+
+void commitAndPushChanges(String version) {
+  runSafely 'git config --global push.default simple'
+  sonatypeZionGitConfig()
+  sshagent(credentials: [sonatypeZionCredentialsId()]) {
+    runSafely 'git add .'
+    runSafely "git diff --exit-code --cached || git commit -m 'Update IQ Server to ${version} for the Red Hat image.'"
+
+    // pull and merge any new commits on main so that the push doesn't fail
+    runSafely 'git pull --no-rebase --no-edit origin main'
+    runSafely 'git push origin HEAD:main'
+  }
+}
+
 node('ubuntu-zion') {
   try {
     stage('Preparation') {
+      githubStatusUpdate('pending')
```

Comment:
If we are going to add this status we should add more statuses to indicate completion

Reply:
the dockerized pipeline does this: 
```
  onSuccess: {
    githubStatusUpdate('success')
  },
  onFailure: {
    githubStatusUpdate('failure')
  }
```

We need to do the equivalent if we add status updates here. Though it may not be as easy since those lifecycle hooks may not be as easily accessible

Reply:
better to keep this in the dockerized pipeline, isn't it?

Reply:
I don't think so. We don't want to introduce the dockerized pipeline since that will build the image. Building the image happens in `build_and_push_rh_image.sh`. So I would say either find a way to give github status checks with out the dockerized pipeline or don't provide any at all (which may not be a good practice).

Reply:
This could work: 2a67362a11cd2eb4e0733017dad7b6f27d60ded0

I only have one question whether I should throw the exception or not. @shaikhu  do you know? What do you think about this approach?

Reply:
> This could work: [2a67362](https://github.com/sonatype/docker-nexus-iq-server/commit/2a67362a11cd2eb4e0733017dad7b6f27d60ded0)
> 
> I only have one question whether I should throw the exception or not. @shaikhu do you know? What do you think about this approach?

@johnatan-velez Usman is out until September, so better check with someone else from BNR team

Reply:
Looks better now. I second Alex's advice of reaching out to BNR for feedback